src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java

```diff
@@ -411,6 +361,125 @@ or at (start + paymentTokenTimeout - offset) where `start` is the creation time
+                                                             TransactionClosureSendCommand command,
+                                                             BigDecimal fee,
+                                                             BigDecimal amount,
+                                                             BigDecimal totalAmount,
+                                                             ClosePaymentRequestV2Dto.OutcomeEnum outcomeEnum
+    ) {
+        return new TransactionDetailsDto()
+                .transaction(
+                        buildTransactionDto(
+                                authRequestData,
+                                transactionAuthorizationRequestData,
+                                transactionAuthorizationCompletedData,
+                                command,
+                                fee,
+                                amount,
+                                totalAmount,
+                                outcomeEnum
+                        )
+                )
+                .info(
+                        buildInfoDto(transactionActivatedData, transactionAuthorizationRequestData)
+                )
+                .user(new UserDto().type(UserDto.TypeEnum.GUEST));
+    }
+
+    private TransactionDto buildTransactionDto(
+                                               AuthRequestDataUtils.AuthRequestData authRequestData,
+                                               TransactionAuthorizationRequestData transactionAuthorizationRequestData,
+                                               TransactionAuthorizationCompletedData transactionAuthorizationCompletedData,
+                                               TransactionClosureSendCommand command,
+                                               BigDecimal fee,
+                                               BigDecimal amount,
+                                               BigDecimal totalAmount,
+                                               ClosePaymentRequestV2Dto.OutcomeEnum outcomeEnum
+    ) {
+        return new TransactionDto()
+                .transactionStatus(
+                        ClosePaymentRequestV2Dto.OutcomeEnum.OK.equals(outcomeEnum) ? "Autorizzato" : "Rifiutato"
+                )
+                .fee(fee)
+                .amount(amount)
+                .grandTotal(totalAmount)
+                .transactionId(
+                        Optional.ofNullable(command.getData()).get().transaction()
+                                .getTransactionId().value()
+                )
+                .creationDate(
+                        Optional.ofNullable(command.getData()).get().transaction()
+                                .getCreationDate().toOffsetDateTime()
+                )
+                .paymentGateway(
+                        Optional.ofNullable(transactionAuthorizationRequestData).get().getPaymentGateway().name()
```

Comment:
why do you check for the correct value of transactionAuthorizationRequestData ? 
```suggestion
                        transactionAuthorizationRequestData.getPaymentGateway().name()
```